### PR TITLE
docs: align config descriptions

### PR DESCRIPTION
**Pull request by bmorelli25**
* This PR moves some text around so that every .NET agent config value follows the same format:
  1. Description
  1. env/IConfig
  1. default/type
* Fixes `added[]` formatting by removing backticks from two headings
* Fixes a tiny alphabetization bug in the options summary

Closes https://github.com/elastic/apm-agent-dotnet/issues/876.

### CODE REVIEW

**Review by gregkalapos — APPROVED** [review]
LGTM

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -186,7 +186,7 @@ A boolean specifying if the agent should be recording or not.
 When recording, the agent captures HTTP requests, tracks errors, and collects and sends metrics.
 When not recording, the agent works as a noop, not collecting data and not communicating with the APM server,
 except for polling the central configuration endpoint.
-As this is a reversible switch, agent threads are not being killed when inactivated, but they will be 
+As this is a reversible switch, agent threads are not being killed when inactivated, but they will be
```
Comment (russcam) [inline docs/configuration.asciidoc:189]:
```suggestion
As this is a reversible switch, agent threads are not killed when inactivated, but they will be
```

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -475,6 +475,10 @@ NOTE: This option requires APM Server 7.2 or later. It will have no effect on ol
 [[config-server-url]]
 ==== `ServerUrl`
 
+The URL for your APM Server. The URL must be fully qualified, including protocol (`http` or `https`) and port.
+
+IMPORTANT: Use of `ServerUrls` is deprecated. Use `ServerUrl`.
```
Comment (russcam) [inline docs/configuration.asciidoc:480]: ++ thanks!

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -507,18 +515,18 @@ IMPORTANT: Use of `ServerUrls` is deprecated. Use `ServerUrl`.
...
-Use this setting in case the APM Server requires a token (e.g. APM Server in Elastic Cloud).
-
-WARNING: the `SecretToken` is sent as plain-text in every request to the server, so you should also secure
-your communications using HTTPS. Unless you do so, your API Key could be observed by an attacker.
-
 [float]
 [[config-api-key]]
 ==== `ApiKey` (added[1.4])
 
+This base64-encoded string is used to ensure that only your agents can send data to your APM server.
```
Comment (russcam) [inline docs/configuration.asciidoc:522]:
```suggestion
A base64-encoded string used to ensure that only your agents can send data to your APM server.
```

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -487,14 +491,18 @@ NOTE: This option requires APM Server 7.2 or later. It will have no effect on ol
...
 
-The URL for your APM Server. The URL must be fully qualified, including protocol (`http` or `https`) and port.
-
-IMPORTANT: Use of `ServerUrls` is deprecated. Use `ServerUrl`.
-
 [float]
 [[config-secret-token]]
 ==== `SecretToken`
 
+This string is used to ensure that only your agents can send data to your APM server.
```
Comment (russcam) [inline docs/configuration.asciidoc:498]:
```suggestion
A string used to ensure that only your agents can send data to your APM server.
```

File: docs/configuration.asciidoc [hunk 5]
```diff
@@ -618,29 +605,41 @@ if there are events that need to be sent out because they were held for too long
...
 | Default                 | Type
-| 1000                    | Integer
+| 10                      | Integer
 |============
 
+[float]
+[[config-max-queue-event-count]]
+==== `MaxQueueEventCount` (added[1.1])
+
 The maximal number of events to hold in the queue as candidates to be sent.
```
Comment (russcam) [inline docs/configuration.asciidoc:624]:
```suggestion
The maximum number of events to hold in the queue as candidates to be sent.
```

**Review by russcam — APPROVED** [review]
LGTM, left some small changes for your perusal

**Pull request merged**